src/Aspire.Hosting/ApplicationModel/CustomResourceSnapshot.cs

```diff
@@ -61,6 +61,11 @@ public ResourceStateSnapshot? State
     /// </summary>
     public int? ExitCode { get; init; }
 
+    /// <summary>
+    /// A flag indicating whether the resource ready event has completed.
+    /// </summary>
+    internal bool ResourceCompletedEventFired { get; init; }
```

Comment:
Why `ResourceCompletedEventFired`? Should it be `ResourceReadyEventFired`?

How would this scale to starting/stopping/restarting a resource multiple times? Should something reset it back to false?

Reply:
I'll rename it. Today these events only fire once.